pkg/docker/config/config.go

```diff
@@ -64,11 +65,17 @@ func GetAuthentication(sys *types.SystemContext, registry string) (string, strin
 	}
 
 	dockerLegacyPath := filepath.Join(homedir.Get(), dockerLegacyCfg)
+	var paths []string
 	pathToAuth, err := getPathToAuth(sys)
-	if err != nil {
-		return "", "", err
+	if err == nil {
+		paths = append(paths, pathToAuth)
```

Comment:
I think you reversed it.  We want pathToAuth to be first.
Does this work?
```
paths = append([]string{pathToAuth}, paths ...)
```

Reply:
yup, it will be the first thing in `paths`. The other file paths are appended later on.

Reply:
Ok I'm an idiot, I thouht you had just taken my example, but you did it differently.

LGTM